cas-server-core-services/src/test/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDaoTests.java

```diff
@@ -0,0 +1,391 @@
+        });
+    }
+
+    @Test
+    public void checkForAuthorizationStrategy() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setServiceId("^https://.+");
+        r.setName("checkForAuthorizationStrategy");
+        r.setId(42);
+
+        final DefaultRegisteredServiceAccessStrategy authz =
+                new DefaultRegisteredServiceAccessStrategy(false, false);
+
+        final Map<String, Set<String>> attrs = new HashMap<>();
+        attrs.put("cn", Sets.newHashSet("v1, v2, v3"));
+        attrs.put("memberOf", Sets.newHashSet(Arrays.asList("v4, v5, v6")));
+        authz.setRequiredAttributes(attrs);
+        r.setAccessStrategy(authz);
+
+        final RegisteredService r2 = this.dao.save(r);
+        final RegisteredService r3 = this.dao.findServiceById(r2.getId());
+        assertEquals(r2, r3);
+    }
+
+    @Test
+    public void verifyAccessStrategyWithStarEndDate() throws Exception {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setServiceId("^https://.+");
+        r.setName("verifyAAccessStrategyWithStarEndDate");
+        r.setId(62);
+
+        final TimeBasedRegisteredServiceAccessStrategy authz =
+                new TimeBasedRegisteredServiceAccessStrategy(true, false);
+
+        authz.setStartingDateTime(ZonedDateTime.now(ZoneOffset.UTC).plusDays(1).toString());
+        authz.setEndingDateTime(ZonedDateTime.now(ZoneOffset.UTC).plusDays(10).toString());
+
+        authz.setUnauthorizedRedirectUrl(new URI("https://www.github.com"));
+        r.setAccessStrategy(authz);
+
+        final RegisteredService r2 = this.dao.save(r);
+        final RegisteredService r3 = this.dao.findServiceById(r2.getId());
+        assertEquals(r2, r3);
+    }
+
+    @Test
+    public void verifyAccessStrategyWithEndpoint() throws Exception {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setServiceId("^https://.+");
+        r.setName("verifyAccessStrategyWithEndpoint");
+        r.setId(62);
+
+        final RemoteEndpointServiceAccessStrategy authz = new RemoteEndpointServiceAccessStrategy();
+
+        authz.setEndpointUrl("http://www.google.com?this=that");
+        authz.setAcceptableResponseCodes("200,405,403");
+        authz.setUnauthorizedRedirectUrl(new URI("https://www.github.com"));
+        r.setAccessStrategy(authz);
+
+        final RegisteredService r2 = this.dao.save(r);
+        final RegisteredService r3 = this.dao.findServiceById(r2.getId());
+        assertEquals(r2, r3);
+    }
+
+    @Test
+    public void serializePublicKeyForServiceAndVerify() throws Exception {
+        final RegisteredServicePublicKey publicKey = new RegisteredServicePublicKeyImpl(
+                "classpath:RSA1024Public.key", "RSA");
+
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setServiceId("^https://.+");
+        r.setName("serializePublicKeyForServiceAndVerify");
+        r.setId(4245);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['4245' is a magic number.](https://www.codacy.com/app/mmoayyed/cas/file/2644137382/issues/source?bid=3256115&fileBranchId=3322154#l349)